packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -571,33 +581,37 @@ export class Relayer {
     level = "info"
   ): Promise<{
     txStatus: boolean;
-    receipt: TransactionReceipt | null;
-    transactionConfig: AugmentedSendOptions | null;
+    executionResult: executedTransaction | null;
   }> {
     try {
       await this.gasEstimator.update();
-      const { receipt, transactionConfig } = await runTransaction({
+      // Run the transaction provided. Note that waitForMine is set to false. This means the function will return as
+      // soon as the transaction has been included in the mem pool, but is not yet mined. This is important as we want
+      // to be able to fire off as many transactions as quickly as posable. Note that as soon as the transaction is
+      // in the mem pool we will produces a transaction hash for logging.
+      const executionResult = await runTransaction({
         web3: this.l1Client.l1Web3,
         transaction,
         transactionConfig: { ...this.gasEstimator.getCurrentFastPrice(), from: this.account },
         availableAccounts: 1,
+        waitForMine: false,
       });
-      if (receipt) {
+      if (executionResult.receipt) {
         this.logger.log({
           level,
           at: "AcrossRelayer#TxProcessor",
           message,
-          mrkdwn: mrkdwn + " tx: " + createEtherscanLinkMarkdown(receipt.transactionHash),
+          mrkdwn: mrkdwn + " tx: " + createEtherscanLinkMarkdown(executionResult.transactionHash),
         });
-        return { txStatus: true, receipt, transactionConfig };
-      } else throw receipt;
+        // Just because the transaction was successfully included in the mem pool does not mean it will be mined without
+        // reverting. Store the transaction execution result within the executedTransactions array. This is processed
+        // at the end of the bot execution loop to ensure that all submitted transactions were successfully included.
+        this.executedTransactions.push(executionResult);
```

Comment:
this is the only part of this file that sends transactions and therefore the only place where we need to append transaction results :)